src/scripts/lang-redirects.js

```diff
@@ -42,7 +42,8 @@ export function handleLanguageBasedRedirects() {
 		ex: /mybranch/myfeature/index.html => /mybranch/myfeature/ja/index.html
 	*/
 	if ( subdomain.includes('preview') || subdomain.includes('docs-staging') ) {
-		previewPath = uri.split('/').slice(0,3).join('/');
+		const commitRef = document.querySelector('html').getAttribute('data-commit-ref'); 
```

Comment:
think this is cleaner

```suggestion
		const commitRef = document.documentElement.dataset.commitRef;
```